contrib/registry/nacos/watcher.go

```diff
@@ -41,10 +41,15 @@ func newWatcher(ctx context.Context, cli naming_client.INamingClient, serviceNam
 		Clusters:    clusters,
 		GroupName:   groupName,
 		SubscribeCallback: func(services []model.SubscribeService, err error) {
-			w.watchChan <- struct{}{}
+			if len(w.watchChan) == 0 {
+				w.watchChan <- struct{}{}
+			}
```

Comment:
```go
		select {
		case w.watchChan <- struct{}{}:
		default:
		}
```
It seems that using `select` would be a better choice

Reply:
The effect is the same, I can accept it all. Do you need me to modify it?

Reply:
I have switched to the recommended implementation and passed the test locally